### DISCUSSION TITLE
Planned new Features

### DISCUSSION THREAD

**Discussion by mcordes92** [eduardogsilva/wireguard_webadmin#20] [General]
Are there currently planned functions that will be implemented in the near future?

**Comment by eduardogsilva** [comment]
Hello there! 

At this moment, I'm planning to implement an import tool, a better dashboard and the ability to "drag and drop" for reordering of peers and firewall rules.
I also would love to hear about suggestions of new features. 
Do you have any requests?

**Reply by mcordes92** [reply]
Maybe a dashboard overview where you can see the following information via graphics / charts, number of instances (online / offline) number of peers (online / offline) traffic that is in and out of the WAN and perhaps the traffic in the individual instances

**Reply by mcordes92** [reply]
And since I'm a monitoring fanatic, an interface for Prometheus Metrics would be nice if something like that is possible

**Reply by eduardogsilva** [reply]
I've spent some time trying to make a nice grafana dashboard to use as an addon. The idea was to include an optional grafana (and maybe prometheus) on the docker compose ready to be used. 

Currently I'm working on another open source project (router management and backup), but I will come back soon to implement the dashboard and the import tool. ;)

**Reply by mcordes92** [reply]
Would a DNS server be possible for installation and administration?

**Comment by eduardogsilva** [comment]
> Would a DNS server be possible for installation and administration?

Hello @mcordes92 , can you please describe your required scenario? 

I'm not quite sure if it matches the project. but anyway, maybe you can just install a dns resolver like bind on the main host or any other docker project like pihole with some extra settings for a fixed private ip. (Just don't leave a resolver open to the internet).

**Reply by mcordes92** [reply]
It was just my question here, I tried to pass on a DNS server to the VPN instances via docker so that I could use the DNS server in the VPN, but this was not possible. That's why I'm asking whether something like this would be possible to install directly into your system

**Reply by mcordes92** [reply]
The scenario is currently as follows: I have two instances, one for all servers and one for all clients (e.g. mobile phones, workstations, etc.). Now I would like to set DNS entries for an internal domain that I can then use via the VPN

**Reply by eduardogsilva** [reply]
I'm glad to see the ability to use multiple wireguard instances being used properly. 

I'm finishing a larger project that will be integrated to wireguard_webadmin in the next few days. After the initial release, I will take a look on the possibility of integrating a DNS server. There are a few technical challenges that need to be resolved depending on the implementation. 

If you try to setup another docker instance for dns server, it will not work out of the box due to how docker networking isolation works unless you configure it properly. 

As a suggestion for a "kind of easy" workaround (without messing with docker network), you could create a private ip address on any existing or virtual network interface on host machine, and run the docker image binding the udp53 port to this private address.

Just take a look on the default firewall rules for wireguard_webadmin, maybe there is a rule blocking the exit traffic to private networks. In this case, you need to remove this rule or add a new rule allowing traffic "before" the reject rule.

**Reply by mcordes92** [reply]
Thanks for the message, I would be happy about an integration if you need any support there or I would be happy to make someone available to test it

**Reply by eduardogsilva** [reply]
Hello @mcordes92 sorry for taking so long, but I just integrated a DNS service to wireguard_webadmin. 

Please take a look at https://github.com/eduardogsilva/wireguard_webadmin/discussions/22

Cheers!

**Comment by xgusto** [comment]
I already mentioned it in another discussion.
It would be nice to maintain rx and tx values even after a reboot or wireguard restart.

**Reply by eduardogsilva** [reply]
This is interesting, but it's hard to keep track of the rx/tx for each peer on wireguard. 
Any kind service restart would reset the counters. If you have any suggestion of how to do this in a reliable way that won't require a lot of time invested, neither high cpu/memory usage, open an issue dedicated for this.

Thanks!

**Comment by xgusto** [comment]
I'm not a programmer, so I can't solve this. Maybe it would be good to store rx and tx in DB at certain intervals.

**Reply by eduardogsilva** [reply]
As discussed previously [here](https://github.com/eduardogsilva/wireguard_webadmin/discussions/14) unfortunately is not that straight forward. I will check this in a later time.

Cheers

**Reply by xgusto** [reply]
It would be great if that function existed. However, it is not necessary. wireguard_webadmin is still a great wg manager.

**Comment by scotticles** [comment]
would like some way to push logs to a syslog server

**Reply by eduardogsilva** [reply]
Hello, I believe that you can use the docker daemon for pushing logs to a remote syslog.

**Comment by eduardogsilva** [comment]
Hey there! Just to give you guys a heads up. 

I just released an update  that uses rrdtool to keep peer traffic information permanent.
Take a look here: https://github.com/eduardogsilva/wireguard_webadmin/discussions/49

Cheers!!
